src/main/java/org/sonar/plugins/csharp/CSharpSensor.java

```diff
@@ -208,53 +218,46 @@ private static String escapeXml(String str) {
+    for (Entry<String, String> param : activeRule.params().entrySet()) {
       builder.put(param.getKey(), param.getValue());
     }
 
-    for (RuleParam param : activeRule.getRule().getParams()) {
-      if (!builder.containsKey(param.getKey())) {
-        builder.put(param.getKey(), param.getDefaultValue());
-      }
-    }
-
     return ImmutableMap.copyOf(builder);
   }
 
   private void importResults(SensorContext context) {
-    File analysisOutput = toolOutput();
+    File analysisOutput = toolOutput(context.fileSystem());
 
-    new AnalysisResultImporter(context, fs, fileLinesContextFactory, noSonarFilter, perspectives).parse(analysisOutput);
+    new AnalysisResultImporter(context, fileLinesContextFactory, noSonarFilter).parse(analysisOutput);
   }
 
-  private void importRoslynReport(String reportPath, final Issuable projectIssuable) {
-    ImmutableMultimap<String, ActiveRule> activeRoslynRulesByPartialRepoKey = RoslynProfileExporter.activeRoslynRulesByPartialRepoKey(ruleProfile.getActiveRules());
+  private void importRoslynReport(String reportPath, final SensorContext context) {
+    ImmutableMultimap<String, RuleKey> activeRoslynRulesByPartialRepoKey = RoslynProfileExporter.activeRoslynRulesByPartialRepoKey(context.activeRules()
+      .findAll()
+      .stream()
+      .map(r -> r.ruleKey())
+      .collect(toList()));
     final Map<String, String> repositoryKeyByRoslynRuleKey = Maps.newHashMap();
-    for (ActiveRule activeRoslynRule : activeRoslynRulesByPartialRepoKey.values()) {
-      String previousRepositoryKey = repositoryKeyByRoslynRuleKey.put(activeRoslynRule.getRuleKey(), activeRoslynRule.getRepositoryKey());
+    for (RuleKey activeRoslynRuleKey : activeRoslynRulesByPartialRepoKey.values()) {
+      String previousRepositoryKey = repositoryKeyByRoslynRuleKey.put(activeRoslynRuleKey.rule(), activeRoslynRuleKey.repository());
       if (previousRepositoryKey != null) {
-        throw new IllegalArgumentException("Rule keys must be unique, but \"" + activeRoslynRule.getRuleKey() +
-          "\" is defined in both the \"" + previousRepositoryKey + "\" and \"" + activeRoslynRule.getRepositoryKey() +
+        throw new IllegalArgumentException("Rule keys must be unique, but \"" + activeRoslynRuleKey.rule() +
+          "\" is defined in both the \"" + previousRepositoryKey + "\" and \"" + activeRoslynRuleKey.repository() +
           "\" rule repositories.");
       }
     }
 
-    SarifParserCallback callback = new ParserCallback(repositoryKeyByRoslynRuleKey, projectIssuable);
-    SarifParser parser = SarifParserFactory.create(new File(reportPath));
-    parser.parse(callback);
+    SarifParserCallback callback = new SarifParserCallbackImpl(context, repositoryKeyByRoslynRuleKey);
+    SarifParserFactory.create(new File(reportPath)).parse(callback);
   }
 
-  private class ParserCallback implements SarifParserCallback {
+  private class SarifParserCallbackImpl implements SarifParserCallback {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this a "static" inner class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2694)